services/madoc-ts/src/frontend/site/pages/loaders/canvas-loader.tsx

```diff
@@ -29,6 +30,15 @@ export const CanvasLoader: UniversalComponent<CanvasLoaderType> = createUniversa
       [data]
     );
 
+    if (!data || data.canvas.source_id === 'not-found') {
```

Comment:
When the data is loading, `data` will be undefined, so as you are navigating it may show "not found" and then jump to a found page. 

the `userData` does return a status. I think this should only be used _if_ we know it's an error (so data source id matching, or status is 'error')

If we don't know, and it's still loading, we can optimistically render what we had before.

The reason is that `<AutoSlotLoader />` will kick off requests to get the slots, and `<Outlet />`  will also allow other loaded to run in parallel. For example, if you're viewing a canvas within a project, the loaders will look something like this:

```jsx
<ProjectLoader>
  <ManifestLoader>
    <CanvasLoader>
       ...
    </CanvasLoader>
  </ManifestLoader>
</ProjectLoader>
```

When that page is first loaded, because all the components optimistically render the `<Outlet/>` they all fire their requests in parallel for the project, manifest and canvas.

If we don't render the `<Outlet/>`, then it will waterfall. So first it will wait for the project to be loaded (as `data === undefined`):
```jsx
<ProjectLoader>
  ...
</ProjectLoader>
```
And then the manifest:
```jsx
<ProjectLoader>
  <ManifestLoader>
    ...
  </ManifestLoader>
</ProjectLoader>
```
And so on.